### ISSUE TITLE
Feature request: An 'alert' system to detect keywords from unwanted posts

### ISSUE THREAD

**Issue by tclaus** [diaspora/diaspora#8259]
As a podmin I regular check my local posts (thanks to #8232 ) for unwanted / suspicious posts that violates common diasporas' terms of use.  (Other user don't report such posts, that does not mean they agree with its content) 
A function that might auto-report posts with keywords per Post to the reporting-system. Ten podmin can decide to handle it by removing or to ignoring this. 
A table with suspicious keywords should then be added and every incoming post will be scanned for existing keywords. Its then added to the reports-list (flagged with a 'suspicious'), to re-check. 
Best: auto-remove from this list after a period of time, if not handled. 

This would also need a UI to manage the keywords. (CRUD). 
Any ideas / comments?

**Comment by Flaburgan** [comment]
I would probably not use such features but I see some podmins who could be interested by it yeah.

**Comment by weex** [comment]
There are many possible solutions but I wanted to try to restate the problem in its simplest form.

    Removing spam takes too much time because messages must be flagged manually.

Does that capture it? Since you only want the messages flagged there isn't much problem with false positives of a simple keyword match.

**Comment by tclaus** [comment]
Not exactly. Spam I only one aspect of "unwanted" posts, its a subset of posts I don't want to see. 
So a voll solution might (will?) use some vectors to detect unwanted posts:
- User reports
- Keyword-Alerts
- Spam-detection facilities (interns, external ones (Askimet?)

My origin Idea only addresses a simple detection by keywords. If implemented a modularized design should be in mind then.

Should this be transfers to discourse?

**Comment by weex** [comment]
If limited to keywords, then I think it's sufficiently described above.

**Comment by tclaus** [comment]
Done on a forked branch.
Will not create a PR on main diaspora branch for this. Too much implications here.

**Comment by denschub** [comment]
"A fork as implemented it" does not invalidate an upstream feature request. Now, granted, this is probably the lowest priority of a feature request, and I have severe stomach pain even thinking about the implications here, but my thoughts and your fork work on its own do not invalidate a feature request. Reopening.

**Comment by tclaus** [comment]
Totally agree, but I don't not want to keep an issue open, I do not request any more. Keep it open.
